src/OpenTelemetry.Api/Trace/TelemetrySpan.cs

```diff
@@ -301,7 +301,10 @@ public string GetBaggageItem(string key)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public TelemetrySpan AddBaggage(string key, string value)
         {
-            this.Activity?.AddBaggage(key, value);
+            if (this.IsRecording)
```

Comment:
its best to not touch this. as Mikel's PR is removing this. :)

Reply:
ok. will revert this change.

Reply:
done.